nocasedict/_nocasedict.py

```diff
@@ -37,85 +37,86 @@
+            from the positional arguments will be empty.
 
-          * If no positional argument is provided, or if one argument with the
-            value `None` is provided, the new dictionary will be empty in this
-            step.
+            For mapping objects, the dictionary will be initialized with the
+            key-value pairs from the mapping object, preserving the iteration
+            order of the mapping object.
 
-          * If one positional argument of tuple or list type is provided, the
-            items in that iterable must be tuples of key and value. These
-            key/value pairs will be added to the new dictionary without copying
-            them, preserving their order in the list.
+            For iterable objects, the dictionary will be initialized with the
+            items from the iterable in iteration order. If a key occurs more
+            than once (case-insensitively), the last value for that key becomes
+            the corresponding value in the dictionary. Each item in the iterable
+            must be one of:
 
-          * If one positional argument of dictionary kind (dict, OrderedDict,
-            NocaseDict) is provided, its key/value pairs are added to the new
-            dictionary without copying them, preserving their order in case of
-            OrderedDict or NocaseDict. Because dict types inherently do not
-            preserve order, the resulting order of items in the new dictionary
-            will be arbitrary in that case.
+            * an iterable with exactly two items. The first item is used as the
+              key, and the second item as the value.
 
-          * Otherwise, `TypeError` is raised.
+            * an object with a ``name`` attribute. The value of the ``name``
+              attribute is used as the key, and the object itself as the value.
 
-        In the second step, the new dictionary is updated from any provided
-        keyword arguments, without copying them, for each keyword argument
-        using its name as a key and its value as a value. Note that Python
-        before version 3.6 uses standard dict objects for passing keyword
-        arguments, so the resulting order of items in the new dictionary will
-        be arbitrary. From Python 3.6 on, keyword arguments will be passed
-        in the order specified, so they will be added to the new dictionary
-        in the order specified, from left to right.
+          **kwargs : Optional keyword arguments representing key-value pairs to
+            add to the dictionary initialized from the positional argument.
 
-        Summary w.r.t. preservation of item order: In order to preserve the
-        order of items used to initialize a new NocaseDict object, only the
-        following approaches can be used across all Python versions supported
-        by this package:
+            Starting with Python 3.7, the order of keyword arguments
```

Comment:
Don't we do it for all python versions.

Reply:
That is what the dict does that is used to actually pass the kwargs in Python. We cannot influence that. And in Python 3.7, the ordering started to be guaranteed.

Reply:
Discussed and agreed.